### PR TITLE
call bam_flagstats autoqc check via the qc script

### PR DESCRIPTION
**Pull request by mgcam**
Depends on https://github.com/wtsi-npg/npg_qc/pull/357

### CODE REVIEW

File: lib/npg_seq_melt/merge.pm [hunk 1]
```diff
@@ -121,14 +124,14 @@ sub run_cmd {
...
-        };
-    return 1;
+
+    my $err = 0;
+    if (system "$start_cmd") {
+        $err = $CHILD_ERROR >> $ERROR_VALUE_SHIFT;
+        $self->log(qq[System command ***$start_cmd*** failed with error $err]);
+    }
+
+    return $err;
```
Comment (jidur) [inline lib/npg_seq_melt/merge.pm]: Maybe just change this to 'return $err ? 0 : 1'  to keep consistant with the original, which returned 0 if error. We could ossibly do a separate pull request later which includes updated calls to run_cmd which could handle the error code being returned?

Comment (mgcam) [inline lib/npg_seq_melt/merge.pm]: Agree, fixed, see https://github.com/wtsi-npg/npg_seq_melt/pull/24/commits/6e79a8ac4f91702bedc27f1156cb5f1e1f9eb87d

File: lib/npg_seq_melt/merge/qc.pm [hunk 2]
```diff
@@ -40,32 +40,31 @@ sub _build_merged_qc_dir {
...
+    my $command = q[];
+    foreach my $arg ( sort keys %{$args} ) {
+      $command .= q[ --] . $arg . q[ ] . $args->{$arg};
+    }
+    $command = 'qc ' . $command;
+    if (!$self->run_cmd($command)) {
+      croak 'QC script exited';
+    }
+
+    return;
```
Comment (jidur) [inline lib/npg_seq_melt/merge/qc.pm]: I think this should return 1 (for now) as library.pm is expecting a true value for success

Comment (mgcam) [inline lib/npg_seq_melt/merge/qc.pm]: fixed, see https://github.com/wtsi-npg/npg_seq_melt/pull/24/commits/f738f929e86830670fcb69b41515e99e92563006

**Pull request merged**